bin/completed_segments.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python
+""" Find (and delete) fully completed kafka-backup segments """
+
+import os
+import re
+import sys
+import argparse
+
+
+def filename_pattern():
+    """ Kafka-backup segment/index filename pattern """
+    return 'segment_partition_%s_from_offset_%s_%s'
+
+
+def parse_args():
+    """ Parse cmdline args """
+    parser = argparse.ArgumentParser(
+        description='Find (and delete) fully completed kafka-backup segments'
+    )
+    parser.add_argument(
+        '-d', '--delete',
+        help='Delete fully completed segment files',
+        action='store_true',
+        default=False,
+    )
+    parser.add_argument(
+        'target_dir',
+        help='Backup directory (target.dir)',
+        default=os.getcwd(),
+    )
+    return parser.parse_args()
+
+
+def collect_segments(target_dir):
+    """ Collect segment partitions & offsets per directory (topic) """
+    # Make regex from pattern
+    fregex = re.compile(r"^%s$" % (filename_pattern() % (r'(\d{3})', r'(\d{10})', 'records')))
```

Comment:
woahh… thats complicated^^ I see where you are heading but any idea how to simplify that?

Reply:
The only way to simplify is to use something like `r'^' + ( ...) + '$'`.
I'd prefer to have single place to change filename pattern than easy to read code in this particular case... easiest way is to duplicate pattern as regex..

Reply:
then lets keep it this way ;) The comment above should remind the reader that we are building a regex here.

Reply:
I'll add few more words around to explain why done this way :)